snapshot/src/main/java/io/camunda/zeebe/snapshots/ConstructableSnapshotStore.java

```diff
@@ -19,9 +19,9 @@
    * @param term the term to which the snapshots corresponds to
    * @param processedPosition the processed position in the snapshot
    * @param exportedPosition the exported position in the snapshot
-   * @return an optional with a transient snapshot if new transient snapshot was taken successfully,
-   *     otherwise return an empty optional
+   * @return an Either that contains and exception or transientSnapshot if it was taken
+   *     successfully.
    */
-  Optional<TransientSnapshot> newTransientSnapshot(
+  Either<SnapshotException, TransientSnapshot> newTransientSnapshot(
```

Comment:
:thought_balloon: First I thought why not just return the message (string), but I think this is necessary because in some cases you want to handle it differently?

Reply:
Yes. That is the idea. If snapshot already exists, it is not an error. So AsyncSnapshotDirector can just log it as debug and ignore it. Anything else could indicate a real error.